lib/crawler.js

```diff
@@ -1656,6 +1629,65 @@ Crawler.prototype.wait = function() {
 /*
     Public: Given a function, this method adds it to an internal list maintained
     by the crawler to be executed against each URL to determine whether it should
+    be downloaded or not.
+
+    callback -  Function to be called when evaluating resource headers. This
+                function is passed a queueItem, including stateData properties.
+                It can determine whether it should be requested or not by
+                returning a boolean - false for no, true for yes.
+
+    Examples
+
+        crawler.addDownloadCondition(function(queueItem) {
+            return (queueItem.host !== "evildomain.com");
+        });
+
+    Returns the ID of the fetch condition - used for removing it from the
+    crawler later.
+
+*/
+Crawler.prototype.addDownloadCondition = function(callback) {
+    if (!(callback instanceof Function)) {
+        throw new Error("Download condition must be a function");
+    }
+
+    this._downloadConditions.push(callback);
+    return this._downloadConditions.length - 1;
+};
+
+/*
+    Public: Given the ID of an existing download condition, this function
+    removes it from the crawler's internal list of conditions.
+
+    index - ID of fetch condition to be removed.
+
+    Examples
+
+        crawler.removeDownloadCondition(3);
+
+    Returns true if the fetch condition was removed, and throws an error if it
+    could not be found.
+
+*/
+Crawler.prototype.removeDownloadCondition = function(index) {
+    var crawler = this;
+
+    if (index instanceof Function) {
```

Comment:
Hmm, you're right, indexOf should work fine. Don't know what I was thinking here :/